src/main/java/org/elasticsearch/discovery/zen/publish/PublishClusterStateAction.java

```diff
@@ -171,7 +200,70 @@ public void handleException(TransportException exp) {
         }
     }
 
+    private void resendFullClusterState(final ClusterState clusterState, final DiscoveryNode node, final AtomicBoolean timedOutWaitingForNodes, final BlockingClusterStatePublishResponseHandler publishResponseHandler) {
```

Comment:
can we have unittests for this stuff I think it's pretty much only tested in integration tests. We should abstract stuff away to make it unittestable!